AlarmPlus/Views/AlarmSettingsViewController.swift

```diff
@@ -0,0 +1,182 @@
+        guard let buttonPressed = sender as? UIButton else {
+            return
+        }
+        if buttonPressed.isSelected {
+            //The day that was selected is currently active in the alarm
+            
+            buttonPressed.isSelected = false
+            if alarmCurrentDays.contains(buttonPressed.tag) {
+                let fAlarmCurrentDays = alarmCurrentDays.filter {$0 != buttonPressed.tag}
+                alarmCurrentDays = fAlarmCurrentDays
+            }
+        } else {
+            //The day is pressed and currently is not active
+            
+            buttonPressed.isSelected = true
+            if alarmCurrentDays.contains(buttonPressed.tag) == false {
+                alarmCurrentDays.append(buttonPressed.tag)
+            }
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupView()
+        
+        if alarm == nil {
+            setupNewAlarmInfo()
+        } else {
+            setupAlarmInfo()
+        }
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        
+    }
+    
+    
+    func setupView() {
+        let saveButton = UIBarButtonItem.init(title: "Save", style: .plain, target: self, action: #selector(rightButtonAction(sender:)))
+        navigationItem.rightBarButtonItem = saveButton
+        daySelectButtonsArray = [sundayButton, mondayButton, tuesdayButton, wednesdayButton, thursdayButton, fridayButton, saturdayButton, sundayButton]
+    }
+    
+    @objc func rightButtonAction(sender: UIBarButtonItem) {
+        //Save UIBarbutton item actions
+        var alarmsArray = MemoryFunctions.getSavedAlarmsArray()
+        
+        guard let currentAlarm = alarm else {
+            return
+        }
+        
+        let (setHour, setMinute) = getUserSetTime(datePicker: datePicker)
+        
+        currentAlarm.schedule.setAlarmTime(hour: setHour, minute: setMinute)
+
+        
+        currentAlarm.schedule.setDayaActive(daysArray: alarmCurrentDays)
+
+        
+        currentAlarm.schedule.setActiveAlarms()
+        
+        alarmsArray.append(currentAlarm)
+
+//        let data = MemoryFunctions.archiveWithKeyedArchiver(object: alarmsArray)
+//
+//        do {
+//            print(try MemoryFunctions.unarchiveAlarmArrayData(data))
+//        } catch {
+//            print(error.localizedDescription)
+//        }
+        
+        MemoryFunctions.saveAlarmsToUserDefaults(alarmsArray)
+
+    }
+    
+    //Function gets the information from existing alarm and pupulates it on the view
+    func setupAlarmInfo() {
+        
+    }
+    
+    //Function creates a new alarm object and sets it with some default data
+    func setupNewAlarmInfo() {
+        alarm = Alarm(snooze: Snooze(), alert: Alert(), schedule: Schedule())
+        
+        let calendar = Calendar(identifier: .gregorian)
+        let date = Date()
+        var dateCompenents = calendar.dateComponents(in: .current, from: date)
```

Comment:
Since you aren't using the local `date` variable anywhere else in this function, it's okay to delete the 
`let date = Date()` line and use `var dateComponents = calendar.dateComponents(in: .current, from: Date())`